Userland/Libraries/LibJS/Runtime/Temporal/ISO8601.cpp

```diff
@@ -786,6 +1004,120 @@ bool ISO8601Parser::parse_time_spec()
     return true;
 }
 
+// https://tc39.es/proposal-temporal/#prod-TimeHourMinuteBasicFormatNotAmbiguous
+bool ISO8601Parser::parse_time_hour_minute_basic_format_not_ambiguous()
+{
+    // TimeSpecWithOptionalTimeZoneNotAmbiguous :
+    //     TimeHourNotValidMonth TimeMinute
+    //     TimeHour TimeMinuteNotValidDay
+    //     TimeHourNotThirtyOneDayMonth TimeMinuteThirtyOneOnly
+    //     TimeHourTwoOnly TimeMinuteThirtyOnly
+    {
+        StateTransaction transaction { *this };
```

Comment:
This could have also been implemented in the form of:
```
StateTransaction transaction { *this };
if (parse_time_hour_not_valid_month()) {
	...
} else if (parse_time_hour()) {
	...
} else if (parse_time_hour_not_thirty_one_day_month()) {
	...
} else if (parse_time_hour_two_only()) {
	...
} else {
	return false;
}
```
To match the other productions, but this is also a valid (and might even be a bit more readable) so both look good to me

Reply:
That's the thing, I don't think this can be implemented like the other productions. What happens if we match `TimeHourNotValidMonth` but then don't find a `TimeMinute`? We need to go back and also check `TimeHour`. Likewise with `TimeHourNotThirtyOneDayMonth` and `TimeHourTwoOnly`.

Reply:
Ah, I hadn't considered that some strings would match against both initial productions, this is the best way then.